### PR TITLE
OLS-498: Add user data collector sidecar to deployment

### PR DESCRIPTION
**Pull request by onmete**
## Description

 Add user data collector sidecar to deployment.

## Type of change

- [x] New feature

## TODO
- [ ] don't merge before https://github.com/openshift/lightspeed-service/pull/584

### CODE REVIEW

**Comment by raptorsun** [comment]
The modifications look good :) 
could you add some verification of this new container in the [unit test code ](https://github.com/openshift/lightspeed-operator/blob/main/internal/controller/ols_app_server_assets_test.go#L192) ?

**Comment by onmete** [comment]
Added deployment verification test.

**Review by raptorsun — APPROVED** [review]
/lgtm

File: internal/controller/ols_app_server_deployment.go [hunk 1]
```diff
@@ -189,6 +189,24 @@ func (r *OLSConfigReconciler) generateOLSDeployment(cr *olsv1alpha1.OLSConfig) (
...
+									Name:  "OLS_USER_DATA_PATH",
+									Value: OLSUserDataMountPath,
+								},
+								{
+									Name:  "INGRESS_ENV",
+									Value: "prod",
+								},
+							},
+							Command:   []string{"python3.11", "/app-root/ols/user_data_collection/data_collector.py"},
+							Resources: *resources,
```
Comment (xrajesh) [inline internal/controller/ols_app_server_deployment.go:208]: We need to find a sweet spot for this resource requests. May not need as much as the OLS service

Comment (xrajesh) [reply internal/controller/ols_app_server_deployment.go:208]: We should also use the same SecurityContext as the api container.

**Comment by xrajesh** [comment]
/hold
Holding until we setup the default feedback setting and pod resources is fixed. @onmete

**Comment by bparees** [comment]
```
  [1mSTEP:[0m check HTTPS Post on /v1/query endpoint [38;5;243m@ 04/24/24 09:41:17.656[0m
  2024-04-24T09:41:17Z	INFO	E0424 09:41:17.657036    7809 portforward.go:394] error copying from local connection to remote stream: read tcp4 127.0.0.1:33479->127.0.0.1:40438: read: connection reset by peer
  2024-04-24T09:41:17Z	INFO	E0424 09:41:17.658756    7809 portforward.go:381] error copying from remote stream to local connection: readfrom tcp4 127.0.0.1:33479->127.0.0.1:40438: write tcp4 127.0.0.1:33479->127.0.0.1:40438: write: broken pipe
  < Exit [1m[It][0m should activate TLS on service HTTPS port [38;5;243m@ 04/24/24 09:41:27.983 (10.35s)[0m
  ```
  
  that failure is presumably unrelated to this change unless adding the sidecar has somehow broken ols itself.

/retest

**Pull request merged**